src/poetry/utils/extras.py

```diff
@@ -64,3 +64,19 @@ def _extra_packages(package_names: Iterable[str]) -> Iterator[str]:
                     yield dependency_package_name
 
     return _extra_packages(extra_package_names)
+
+
+def str_to_bool(val: str) -> bool:
+    """Convert a string representation of truth to true (1) or false (0).
+
+    True values are 'y', 'yes', 't', 'true', 'on', and '1'; false values
+    are 'n', 'no', 'f', 'false', 'off', and '0'.  Raises ValueError if
+    'val' is anything else.
+    """
+    val = val.lower()
+    if val in ("y", "yes", "t", "true", "on", "1"):
+        return True
+    elif val in ("n", "no", "f", "false", "off", "0"):
+        return False
+    else:
+        raise ValueError(f"invalid truth value {val!r}")
```

Comment:
The `else` is not needed, since all previous branches return something.
A similar logic applies for `elif`, we can use `if` since the previous condition returns something already.
```suggestion
    if val in ("n", "no", "f", "false", "off", "0"):
        return False
    raise ValueError(f"invalid truth value {val!r}")
```